src/holdup/cli.py

```diff
@@ -84,14 +79,52 @@ def __str__(self):
 
 class HttpCheck(Check):
     def __init__(self, url):
-        self.url = url
-
-    def run(self, timeout, _):
-        if hasattr(ssl, 'create_default_context') and 'context' in getargspec(urlopen).args:
-            kwargs = {'context': ssl.create_default_context()}
+        self.do_insecure = False
+        proto, urn = url.split('://', 1)
+        if proto == 'https+insecure':
+            self.do_insecure = True
+            proto = 'https'
+        self.url = '{}://{}'.format(proto, urn)
+
+    def can_create_default_context(self):
```

Comment:
How come the previous feature check was replaced with this version check? Imo this is looking for trouble (eg, pypy doesn't always have same feature parity on version_info).

Reply:
Though supported, the `getargspec(urlopen).args` in python3 does not contain `context`. That is because of a `*` in `urllib.request.urlopen`'s argspec [1]. Whereas, `urllib2.urlopen` does not have that[2]. In my testing, the current logic only worked for python2 and not for python3.

Here's an example snippet that shows the difference in behavior:

```python
try:
    from inspect import getfullargspec as getargspec
except ImportError:
    from inspect import getargspec

try:
    from urllib.request import urlopen
except ImportError:
    from urllib2 import urlopen

import pprint


pprint.pprint(getargspec(urlopen).args)
```

Running the above snippet does this on my machine:

```shell
# Python2
$ python ./test.py 
['url', 'data', 'timeout', 'cafile', 'capath', 'cadefault', 'context']

# Python3
$ python3 ./test.py 
['url', 'data', 'timeout']
```

[1] https://github.com/python/cpython/blob/3.6/Lib/urllib/request.py#L141
[2] https://github.com/python/cpython/blob/2.7/Lib/urllib2.py#L131

Reply:
Right, cause on py3 it's in `kwonlyargs`, not `args`.

Reply:
Ah. I did not know that. What is your recommendation on how this should be handled?

Reply:
@ionelmc i've switched the logic back to using `inspect` instead of version numbers. Please review.